### ISSUE TITLE
Fixed: PHP 8: Get views exposed filters with checkbox working

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#5305]
**Description of the bug**

Currently the default admin content view doesn't work on PHP 8.

**Steps To Reproduce**

1. Set your web to use PHP 8
2. Go to /admin/content
3. If you still see nodes (with a "new" marker) go to these node's pages and then come back to the admin overview
4. Add new nodes, one by one
5. Go back to /admin/content

**Actual behavior**

The result is always empty, if there are no "new" nodes, no matter which filters are set, or if filters are set at all. Only nodes with the "new" marker appear. And they vanish, as soon as you visited their pages.

**Expected behavior**

Correct list of content.

**Additional information**

TLDR: `var_dump('All' == 0)` returns true in PHP prior to version 8, false since 8.

So this check in function accept_exposed_input() worked in previous PHP versions, but is now broken, which is why the filter always gets added to the query.

**Comment by indigoxela** [comment]
Part of https://github.com/backdrop/backdrop-issues/issues/5076

I've not been able to find any related D7 / D8 / D9 issue, but the problem is easy to reproduce in B, as one of the most important admin pages is a view that uses a checkbox in exposed filters.

Here's a [PR](https://github.com/backdrop/backdrop/pull/3794) to test and review. (The temporary switch to PHP 8 will get reverted, of course).

Off topic: we probably have a lot more problems with checkbox(es) and radio(s) values... Didn't dig deeper re those yet.
